app/javascript/CommentSubscription/CommentSubscription.jsx

```diff
@@ -107,16 +110,21 @@ export class CommentSubscription extends Component {
           <Button
             variant="outlined"
             onClick={(_event) => {
-              if (subscribed) {
-                onUnsubscribe(COMMENT_SUBSCRIPTION_TYPE.NOT_SUBSCRIBED);
-                this.setState({
-                  subscriptionType: COMMENT_SUBSCRIPTION_TYPE.ALL,
-                });
+              if (userStatus == "logged-in") {
+                if (subscribed) {
+                  onUnsubscribe(COMMENT_SUBSCRIPTION_TYPE.NOT_SUBSCRIBED);
+                  this.setState({
+                    subscriptionType: COMMENT_SUBSCRIPTION_TYPE.ALL,
+                  });
+                } else {
+                  onSubscribe(subscriptionType);
+                }
+
+                this.setState({ subscribed: !subscribed });
               } else {
-                onSubscribe(subscriptionType);
+                _event.preventDefault();
```

Comment:
I don't think we need this as there's no default behaviour we need to block (e.g. form submit/link follow)